app/src/main/java/org/fossasia/openevent/activities/MainActivity.java

```diff
@@ -42,6 +43,11 @@ protected void onCreate(Bundle savedInstanceState) {
             setupDrawerContent(navigationView);
         }
 
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+            getWindow().setStatusBarColor(getResources().getColor(R.color.color_primary));
```

Comment:
I have seen reddit flow app do it. Essentially it follows this approach:

http://stackoverflow.com/questions/4663752/switching-application-wide-theme-programmatically

But a restart of any activity is required. Or put it in a BaseActivity
before calling setContentView(). But in any case you can still only declare
styles once in XML, so still needs to be part of the app package. So it is
dynamic in a sense, but only dynamic within the declared styles.


On Wed, Jul 1, 2015 at 1:48 AM, Manan Wason <notifications@github.com>
wrote:

> In app/src/main/java/org/fossasia/openevent/activities/MainActivity.java
> <https://github.com/fossasia/open-event-android/pull/47#discussion_r33602029>
> :
>
> > @@ -42,6 +43,11 @@ protected void onCreate(Bundle savedInstanceState) {
> >              setupDrawerContent(navigationView);
> >          }
> >
> > +        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
> > +            getWindow().setStatusBarColor(getResources().getColor(R.color.color_primary));
>
> Yeah I was thinking about this. We would not be able to change the colour
> of every component by obtaining colors from api. Maybe yeah the primary
> colors but I don't know. @championswimmer
> <https://github.com/championswimmer> you said you have worked on
> something like this before. Can you guide here?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/fossasia/open-event-android/pull/47/files#r33602029>.
>



-- 
Mohit Singh KANWAL (Mr)

[image: Facebook] <http://facebook.com/mohitkanwal>  [image: Twitter]
<http://twitter.com/mohitkanwal>